crates/red_knot_test/src/diagnostic.rs

```diff
@@ -2,10 +2,63 @@
 //!
 //! We don't assume that we will get the diagnostics in source order.
 
+use red_knot_python_semantic::types::TypeCheckDiagnostic;
+use ruff_python_parser::ParseError;
 use ruff_source_file::{LineIndex, OneIndexed};
-use ruff_text_size::Ranged;
+use ruff_text_size::{Ranged, TextRange};
+use std::borrow::Cow;
 use std::ops::{Deref, Range};
 
+pub(super) trait Diagnostic: std::fmt::Debug {
```

Comment:
@carljm the trait doesn't implement `Ranged` because I can't implement `Ranged` for `ParseError` (orphan rule). We could consider implementing it for `ParseError` but the benefit seemed marginal.